wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -42,9 +43,9 @@
 from kubernetes.client.rest import ApiException  # type: ignore # noqa: E402
 
 TIMEOUT = 5
-MAX_KUBERNETES_RETRIES = (
-    60  # default 10 second loop time on the agent, this is 10 minutes
-)
+MAX_KUBERNETES_RETRIES = os.environ.get(
+    "WANDB_AGENT_RETRIES", 180
+)  # default 10 second loop time on the agent, this is 30 minutes
```

Comment:
I think it might be more clear if we allow the user to specify a polling interval and a max wait time rather than a polling interval and # retries.